ale/base/base.py

```diff
@@ -194,27 +235,51 @@ def sensor_model_version(self):
 
     @abc.abstractproperty
     def focal2pixel_lines(self):
+        """
+        Returns
+        -------
+        : int
+          Focal plane to detector lines
+        """
         pass
 
     @abc.abstractproperty
     def focal2pixel_samples(self):
+        """
+        Returns
+        -------
+        : int
+          Focal plane to detector samples
+        """
         pass
 
     @abc.abstractproperty
     def pixel2focal_x(self):
+        """
+        Returns
+        -------
+        : int
```

Comment:
This is similar to focal2pixels, except it is 3 coefficients for the transformation from pixels to focal plane coordinates and the coefficients are constant, sample, then line. So, if the pixel coordinate is (sample, line) and the coefficient array F, then the focal plane x coordinate is F[0] + sample * F[1] + line * F[2].